xhprof_lib/utils/xhprof_runs.php

```diff
@@ -107,22 +107,22 @@ public static function getDbClass()
   * 
   *
 CREATE TABLE `details` (
```

Comment:
> as for the size of data columns, id ok was a wrong one but was when I was debugging because data was not being inserted due to the server_id only allowing 3 chars

I think this change needs to be moved out to separate PR where you'll show the INSERT statement that was used and what SQL error happened. I saw nobody complain, that SQL error happens, so I guess there must be something specific in your MySQL server configuration that causes it to happen (e.g. strict mode).